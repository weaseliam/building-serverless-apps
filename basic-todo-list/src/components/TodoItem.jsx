import React from 'react';
import PropTypes from 'prop-types';

const TodoItem = ({ name, done, onToggleDone }) => (
  <li
    style={{ textDecoration: done ? "line-through" : "none" }}
    onClick={onToggleDone}>
    {name}
  </li>
);

TodoItem.propTypes = {
  name: PropTypes.string,
  done: PropTypes.bool,
  onToggleDone: PropTypes.func
};

TodoItem.defaultProps = {
  name: '',
  done: false,
  onToggleDone: () => { }
};

export default TodoItem;
