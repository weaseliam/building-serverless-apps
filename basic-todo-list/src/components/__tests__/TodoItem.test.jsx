import React from 'react';
import renderer from 'react-test-renderer';

import TodoItem from '../TodoItem';

test('TodoItem renders as expected', () => {
  const component = renderer.create(<TodoItem name="Hello World !" />);

  expect(component.toJSON()).toMatchSnapshot();
});
