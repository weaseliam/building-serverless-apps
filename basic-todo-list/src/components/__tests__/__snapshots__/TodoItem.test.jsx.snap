// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders as expected 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  Hello World !
</li>
`;
